### PR TITLE
Dots at random strings

### PR DESCRIPTION
**Pull request by EzAccount**
Dots at random strings added

### CODE REVIEW

File: translate.py [hunk 1]
```diff
@@ -1,11 +1,24 @@
 import re
 from pathlib import Path
-
```
Comment (uburuntu) [inline translate.py:3]: Revert line between builtin and external packages

File: translate.py [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+import numpy as np
 import bs4
```
Comment (uburuntu) [inline translate.py]: Add package in sorted way

File: translate.py [hunk 3]
```diff
@@ -1,11 +1,24 @@
 import re
 from pathlib import Path
-
+import numpy as np
```
Comment (uburuntu) [inline translate.py]: Why numpy, not builtin random?
If numpy, then add requirement in requirements.txt.

File: translate.py [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
 import bs4
 
 
 def translator(string: str) -> str:
-    return string.rstrip('.') + ')'
+    t = np.random.randint(1, 43)
+    pos = string.find('%1$@')
+    if (pos != -1):
```
Comment (uburuntu) [inline translate.py]: Meaningless brackets

File: translate.py [hunk 5]
```diff
@@ -1,11 +1,24 @@
...
+        pos += 4
+        string = string[:pos]+')'+string[pos:]
```
Comment (uburuntu) [inline translate.py]: Not PEP8, add spaces

File: translate.py [hunk 6]
```diff
@@ -1,11 +1,24 @@
...
+    pos = string.find('%1$s')
+    if (pos != -1):
+        pos += 4
+        string = string[:pos]+')'+string[pos:]
+ 
+    if (t != 42):
+        return string.rstrip('.') + ')'
+    else:
+        return string.rstrip('.') + '...)'
```
Comment (uburuntu) [inline translate.py]: Not real ellipsis -- three dots instead

File: translate.py [hunk 7]
```diff
@@ -1,11 +1,24 @@
...
+        pos += 4
+        string = string[:pos]+')'+string[pos:]
+    pos = string.find('%1$s')
+    if (pos != -1):
+        pos += 4
+        string = string[:pos]+')'+string[pos:]
+ 
+    if (t != 42):
+        return string.rstrip('.') + ')'
+    else:
```
Comment (uburuntu) [inline translate.py]: Meaningless else

**Review by uburuntu — CHANGES_REQUESTED** [review]
Do it better

File: translate.py (hunk 4 above)
Comment (EzAccount) [reply translate.py]: explain in more details please

File: translate.py (hunk 6 above)
Comment (EzAccount) [reply translate.py]: I think three dots are more suitable

**Comment by uburuntu** [comment]
Mark as resolved when resolve

File: translate.py (hunk 4 above)
Comment (uburuntu) [reply translate.py]: use `if pos != -1:` instead of `if (pos != -1):`

File: translate.py [hunk 8]
```diff
@@ -2,13 +2,25 @@
...
+        pos += 4
+        string = string[:pos] + ')' + string[pos:]
+    pos = string.find('%1$s')
+    if (pos != -1):
+        pos += 4
+        string = string[:pos] + ')' + string[pos:]
+ 
+    if (t != 42):
+        return string.rstrip('.') + ')'
+    return string.rstrip('.') + '...)'
```
Comment (uburuntu) [inline translate.py]: Replace if and else blocks. Under `if` you should use more rare case

**Review by uburuntu — CHANGES_REQUESTED** [review]

File: translate.py [hunk 9]
```diff
@@ -2,13 +2,26 @@
 from pathlib import Path
 
 import bs4
-
+import numpy as np
 
 def translator(string: str) -> str:
+    t = np.random.randint(1, 43)
```
Comment (uburuntu) [inline translate.py]: why you need numpy for this?
```python
import random
print(random.randint(1, 43))
```

**Review by uburuntu — CHANGES_REQUESTED** [review]

File: translate.py (hunk 4 above)
Comment (uburuntu) [reply translate.py]: not resolved

File: translate.py [hunk 10]
```diff
@@ -1,14 +1,27 @@
...
+        string = string[:pos] + ')' + string[pos:]
+ 
+    if (t == 42):
+        return string.rstrip('.') + '...)'
     return string.rstrip('.') + ')'
+    
 
 
-def translate_xml(text: str) -> str:
+def translate_xml(text: str) -> str:
```
Comment (uburuntu) [inline translate.py]: remove trailing whitespace

**Review by uburuntu — CHANGES_REQUESTED** [review]

File: translate.py [hunk 11]
```diff
@@ -1,11 +1,24 @@
...
+        string = string[:pos] + ')' + string[pos:]
+    pos = string.find('%1$s')
+    if pos != -1:
+        pos += 4
+        string = string[:pos] + ')' + string[pos:]
+ 
+    if t == 42:
+        return string.rstrip('.') + '...)'
     return string.rstrip('.') + ')'
+
```
Comment (uburuntu) [inline translate.py]: why three empty lines, we need two by pep8

**Review by uburuntu — CHANGES_REQUESTED** [review]

**Pull request merged**